pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -61,6 +61,11 @@ class ApproxTimeEvolution(Operation):
     this decomposition is exact for any value of :math:`n` when each term of the Hamiltonian
     commutes with every other term.
 
+    .. warning::
+
+        Two mathematically equivalent :class:`~.Hamiltonian` objects may undergo different time evolutions
+        due to the order in which the observables are stored.
+
```

Comment:
Great! Would you be up for adding this warning to [TrotterProduct](https://xanaduai-pennylane--5137.com.readthedocs.build/en/5137/code/api/pennylane.TrotterProduct.html), which we are steering towards long term as it is more flexible.

Reply:
Hi Thomas,

I just duplicated the warning in `qml.TrotterProduct`. Hope that works =)
<img src="https://github.com/PennyLaneAI/pennylane/assets/32418878/92cd4ebd-9cda-4225-a1d4-f503ca69f905" width="500">

Reply:
Thanks!